### PR TITLE
Add initial Simple Gripper support

### PR DESCRIPTION
**Pull request by rethink-rlinsalata**
(REPLACED BY PR #122 - same code changes)

---

Provides an initial simple interface to the ClickSmart IO plate and gripper.

Includes:
* New SimpleClickSmartGripper class
  - uses "EE Signal Types" interface to reference signals configured for a "tool"/endpoint on the ClickSmart (i.e. "grip", "closed")
* Helper Gripper "Factory" class / fn
  - eases creation of interface for currently attached Gripper
* Fixes for init of IODeviceInterface, in waiting for init state/config
  - state will not be published for non-activated ClickSmart devices
* IOCommand now has a built-in `as_msg()` fn to handle jsonify'ing IO msgs

### CODE REVIEW

File: intera_interface/src/intera_interface/gripper.py [hunk 1]
```diff
@@ -254,3 +261,185 @@ def set_dead_zone(self, dead_zone):
...
+        self.node_state = msg
+        if msg.devices:
+            self.states = msg.devices
+
+    def _node_config_cb(self, msg):
+        self.node_config = msg
+        if msg.devices:
+            self.configs = msg.devices
+
+    def lookupGripperClass(self, ee_type):
```
Comment (IanTheEngineer) [inline intera_interface/src/intera_interface/gripper.py]: Should be "private"

File: intera_interface/src/intera_interface/gripper.py [hunk 2]
```diff
@@ -254,3 +261,185 @@ def set_dead_zone(self, dead_zone):
...
+        EE_CLASS_MAP = dict({
+            "SmartToolPlate": SimpleIOGripper,
+            "ElectricParallelGripper": Gripper,
+            "default": Gripper
+        })
+
+        return EE_CLASS_MAP.get(ee_type, None) or EE_CLASS_MAP["default"]
+
+    def parseConfig(self, config):
```
Comment (IanTheEngineer) [inline intera_interface/src/intera_interface/gripper.py]: Should be private

File: intera_interface/src/intera_interface/gripper_factory.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    IONodeStatus,
+    IOComponentConfiguration
+)
+import intera_dataflow
+from intera_interface import (
+    Gripper,
+    SimpleClickSmartGripper
+)
+
+
```
Comment (IanTheEngineer) [inline intera_interface/src/intera_interface/gripper_factory.py]: What are your thoughts on an external helper function?
```
def get_current_gripper_interface():
  gf = GripperFactory()
  return gf.get_current_gripper_interface()
```

Comment (rethink-rlinsalata) [reply intera_interface/src/intera_interface/gripper_factory.py]: The external helper function is starting to feel better.

So you thinking the usage would be something like:
```
gripper = intera_interface.get_current_gripper_interface()
```

?

Comment (IanTheEngineer) [reply intera_interface/src/intera_interface/gripper_factory.py]: Yes, that's the thought. What do you think from a usability perspective?

File: intera_interface/src/intera_io/io_interface.py [hunk 4]
```diff
@@ -152,7 +161,7 @@ def publish_command(self, op, args, timeout=2.0):
         """
         cmd_time = rospy.Time.now()
         self.cmd_times.append(cmd_time)
-        self.cmd_times = self.cmd_times[-100:]
+        self.cmd_times = self.cmd_times[-100:]  # cache last 100 cmd timestamps
```
Comment (IanTheEngineer) [inline intera_interface/src/intera_io/io_interface.py]: +1 for the comment :)

**Comment by rethink-rlinsalata** [comment]
Closing - Replaced by PR #122 - proper branch source and includes example and Gripper fixes.

**Pull request closed without merging**